### PR TITLE
Patch @actions/cache to use @action/core version 1.10.0

### PR DESCRIPTION
**Pull request by pdotl**
Update @action/cache package to use the version `1.10.0` of @actions/core.

### CODE REVIEW

**Comment by bishal-pdMSFT** [comment]
Doesn't it need a code change to remove usage of `save-state` and `set-output`?

**Comment by bishal-pdMSFT** [comment]
> Doesn't it need a code change to remove usage of `save-state` and `set-output`?

Looks like that is handled internally https://github.com/actions/toolkit/blob/1bc93f3cdf638ff423193f992119be08b7bb0666/packages/core/src/core.ts#L355

**Comment by pdotl** [comment]
> 

The code changes happened on the `actions/cache` side [here](https://github.com/actions/cache/pull/950).

File: packages/cache/package-lock.json [hunk 1]
```diff
@@ -1,625 +1,13 @@
 {
   "name": "@actions/cache",
-  "version": "3.0.4",
-  "lockfileVersion": 2,
```
Comment (Phantsure) [inline packages/cache/package-lock.json:4]: Why did we change lockfile version?

File: packages/cache/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@actions/cache",
-  "version": "3.0.4",
```
Comment (aparna-ravindra) [inline packages/cache/package.json:3]: please update RELEASES.md as well

File: packages/cache/package-lock.json (hunk 1 above)
Comment (pdotl) [reply packages/cache/package-lock.json:4]: This was changed on using `npm version patch` command. Reverting.

**Review by Phantsure — APPROVED** [review]
lgtm

**Pull request merged**